tracer/test/Datadog.Trace.TestHelpers/MockHttpParser.cs

```diff
@@ -75,7 +75,9 @@ public static async Task<MockHttpRequest> ReadRequest(Stream stream)
             }
             while (true);
 
-            var length = long.TryParse(headers.GetValue(ContentLengthHeaderKey), out var headerValue) ? headerValue : (long?)null;
+            var length = headers.GetValue(ContentLengthHeaderKey) is { } contentLength
+                             ? long.Parse(contentLength)
+                             : (long?)null;
```

Comment:
I don't mind the change, but isn't it exactly the same logic as previously?

Reply:
Pretty much. In the new version we treat non-`long` content-lengths as an error instead of silently ignoring them which I think is worthwhile. But mostly I think this change is an accident related to the fact that in a subsequent PR we change to using `headers.GetValue()` so this construction is more useful